pkg/services/ngalert/api/api_ruler_mock.go

```diff
@@ -0,0 +1,291 @@
+			{
+				Name:     "group1",
+				Interval: 60,
+				Rules: []apimodels.ExtendedRuleNode{
+					{
+						GrafanaManagedAlert: apimodels.ExtendedUpsertAlertDefinitionCommand{
+							NoDataState:         apimodels.NoData,
+							ExecutionErrorState: apimodels.AlertingErrState,
+							UpdateAlertDefinitionCommand: ngmodels.UpdateAlertDefinitionCommand{
+								UID:       "UID",
+								OrgID:     1,
+								Title:     "rule 1-1",
+								Condition: "A",
+								Data: []ngmodels.AlertQuery{
+									{
+										Model: json.RawMessage(`{
+												"datasource": "__expr__",
+												"type":"math",
+												"expression":"2 + 2 > 1"
+											}`),
+										RefID: "A",
+										RelativeTimeRange: ngmodels.RelativeTimeRange{
+											From: ngmodels.Duration(time.Duration(5) * time.Hour),
+											To:   ngmodels.Duration(time.Duration(3) * time.Hour),
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						GrafanaManagedAlert: apimodels.ExtendedUpsertAlertDefinitionCommand{
+							NoDataState:         apimodels.NoData,
+							ExecutionErrorState: apimodels.AlertingErrState,
+							UpdateAlertDefinitionCommand: ngmodels.UpdateAlertDefinitionCommand{
+								UID:       "UID",
+								OrgID:     1,
+								Title:     "rule 1-2",
+								Condition: "A",
+								Data: []ngmodels.AlertQuery{
+									{
+										Model: json.RawMessage(`{
+												"datasource": "__expr__",
+												"type":"math",
+												"expression":"2 + 2 == 1"
+											}`),
+										RefID: "A",
+										RelativeTimeRange: ngmodels.RelativeTimeRange{
+											From: ngmodels.Duration(time.Duration(5) * time.Hour),
+											To:   ngmodels.Duration(time.Duration(3) * time.Hour),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return response.JSON(http.StatusAccepted, result)
+}
+
+func (mock RulerApiMock) RouteGetRulegGroupConfig(c *models.ReqContext) response.Response {
+	datasourceID := c.Params(":DatasourceId")
+	mock.log.Info("RouteGetRulegGroupConfig: ", "DatasourceId", datasourceID)
+	namespace := c.Params(":Namespace")
+	mock.log.Info("RouteGetRulegGroupConfig: ", "Namespace", namespace)
+	groupname := c.Params(":Groupname")
+	mock.log.Info("RouteGetRulegGroupConfig: ", "Groupname", groupname)
+	result := apimodels.RuleGroupConfigResponse{
+		RuleGroupConfig: apimodels.RuleGroupConfig{
+			Name:     groupname,
+			Interval: 60,
+			Rules: []apimodels.ExtendedRuleNode{
+				{
+					GrafanaManagedAlert: apimodels.ExtendedUpsertAlertDefinitionCommand{
+						NoDataState:         apimodels.NoData,
+						ExecutionErrorState: apimodels.AlertingErrState,
+						UpdateAlertDefinitionCommand: ngmodels.UpdateAlertDefinitionCommand{
+							UID:       "UID",
+							OrgID:     1,
+							Title:     "something completely different",
+							Condition: "A",
+							Data: []ngmodels.AlertQuery{
+								{
+									Model: json.RawMessage(`{
```

Comment:
Might be good to mock at least one data source query with an expression, as while an expression  by itself could be an alert, pretty pointless outside of testing.

Reply:
Good point; I was planning to do it.